### PR TITLE
improve importance ratio metrics

### PR DESCRIPTION
**Pull request by samsja**
# Improve importance ratio metrics

this PR add a bunch of new metrics regarding importance ratio, namely the max and min, the raw and unraw ratio, as well as the error sum.

It also refactors the way we handle the importance ratio metrics and disentangles it from the loss metrics. It should make it easier if we want to log the full distribution of the logprob ratio. 

screenshot-20250731-142745

### CODE REVIEW

File: src/prime_rl/trainer/train.py [hunk 1]
```diff
@@ -309,7 +326,7 @@ def train(config: TrainerConfig):
         # Maybe clean up weight checkpoint
         weight_ckpt_manager.maybe_clean(progress.step)
 
-        # Optionally, dump memory snapshot
+        # Optionally, dump memorfy snapshot
```
Comment (Jackmin801) [inline src/prime_rl/trainer/train.py]: whos memorfy?

File: src/prime_rl/trainer/loss.py [hunk 2]
```diff
@@ -182,3 +217,67 @@ def shift_logits(logits: Float[Tensor, "batch seq vocab"]) -> Float[Tensor, "bat
     zeros = torch.zeros(B, 1, V, device=logits.device, dtype=logits.dtype)  # (B, 1, V)
     logits = torch.cat([zeros, logits], dim=1)  # (B, L, V)
     return logits
+
+
+class ImportanceRatioMetrics:
```
Comment (mikasenghaas) [inline src/prime_rl/trainer/loss.py:222]: i think we might wanna generalize this pattern, seems very bulky "only" for loggin importance ratio metrics. maybe smth similar in spirit to the metrics averager, just better?

Comment (samsja) [reply src/prime_rl/trainer/loss.py:222]: I would just keep it like this for now and generalize later, agree that refactor could be nice but not need for now

**Review by mikasenghaas — APPROVED** [review]
